lib/Sema/CSFix.h

```diff
@@ -1288,6 +1291,32 @@ class IgnoreContextualType : public ContextualMismatch {
                                       ConstraintLocator *locator);
 };
 
+class RemoveUnnecessaryCoercion : public ConstraintFix {
+  Type fromType;
+  Type toType;
+  
+protected:
+  RemoveUnnecessaryCoercion(ConstraintSystem &cs,
+                            Type fromType, Type toType,
+                            ConstraintLocator *locator)
+  : ConstraintFix(cs, FixKind::RemoveUnnecessaryCoercion, locator, /*isWarning*/ true),
+                  fromType(fromType), toType(toType) {}
+
+  
+public:
+  std::string getName() const override { return "remove unnecessary explicit type coercion"; }
+  
+  Type getToType() const { return toType; }
```

Comment:
Once `ContextualMismatch` is a parent, get{To, From}Type methods are no longer required.